### PR TITLE
Add fee calculation to TxManager

### PR DESCRIPTION
**Pull request by juanbono**
# Add fee calculation to TxManager

The eigensdk-go version of the TxManager includes fee estimation as part of the sendTransaction flow.
This PR adds just that to the TxManager PR.

### CODE REVIEW

File: crates/chainio/txmanager/src/simple_tx_manager.rs [hunk 1]
```diff
@@ -174,6 +185,78 @@ impl<'log> SimpleTxManager<'log> {
         SimpleTxManager::wait_for_receipt(pending_tx).await
     }
 
+    async fn estimate_gas_and_nonce(
+        &self,
+        tx: &TxLegacy,
+    ) -> Result<TransactionRequest, TxManagerError> {
+        let gas_tip_cap = self.provider.get_max_priority_fee_per_gas().await
+        .inspect_err(|err| self.logger.info("eth_maxPriorityFeePerGas is unsupported by current backend, using fallback gasTipCap", &[err]))
```
Comment (dralves) [inline crates/chainio/txmanager/src/simple_tx_manager.rs]: proper indent also wrap to 120 cols

File: crates/chainio/txmanager/src/simple_tx_manager.rs [hunk 2]
```diff
@@ -174,6 +185,78 @@ impl<'log> SimpleTxManager<'log> {
         SimpleTxManager::wait_for_receipt(pending_tx).await
     }
 
+    async fn estimate_gas_and_nonce(
```
Comment (dralves) [inline crates/chainio/txmanager/src/simple_tx_manager.rs:205]: docs

**Review by dralves — APPROVED** [review]
only two minor nits, approving assuming they will be fixed

File: crates/chainio/txmanager/src/simple_tx_manager.rs (hunk 1 above)
Comment (pablodeymo) [reply crates/chainio/txmanager/src/simple_tx_manager.rs]: done

File: crates/chainio/txmanager/src/simple_tx_manager.rs (hunk 2 above)
Comment (pablodeymo) [reply crates/chainio/txmanager/src/simple_tx_manager.rs:205]: done

**Review by pablodeymo — APPROVED** [review]
LGTM

**Pull request merged**